src/features/Character/pages/CharacterPage/CharacterPage.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { VFC } from 'react';
+import { Box, Container, Grid, Heading, HStack, Image, Text, VStack } from '@chakra-ui/react';
+import { ContentWrapper, Error, TagBlock } from '../../../../components';
+import { useRouteParams } from '../../../../hooks/useRouterParams';
+import { useCharacterQuery } from '../../queries';
+import { CharacterRouteParams } from '../../utils/characterRouteParams';
+import characterPlaceholder from '../../../../assets/person.svg';
+import { BBCode } from '../../../../components/BBCode/BBCode';
+import { CharacterGender } from '../../../../models/characters/characterGender';
+import { useExtendedTraitsQuery } from '../../queries/trait';
+import { CharacterDetail, CharacterTraits } from '../../components';
+import { useRelatedVisualNovelsQuery } from '../../../VisualNovel/queries/visualNovel';
+import { CharacterRole } from '../../../../models/characters/characterRole';
+
+/** Character page. */
+export const CharacterPage: VFC = () => {
+  const { id } = useRouteParams<CharacterRouteParams>();
+  const { isLoading, data, error } = useCharacterQuery(Number(id));
+
+  const traitsIds = data?.traits ?? [];
+  const {
+    isLoading: isTraitsLoading,
+    data: traitsWithRoot,
+    error: traitsError,
+  } = useExtendedTraitsQuery(id, traitsIds, {
+    enabled: traitsIds.length > 0,
+  });
+
+  const visualNovelIds = data?.visualNovels.map(vn => vn.visualNovelId) ?? [];
+  const {
+    isLoading: isVisualNovelsLoading,
+    data: visualNovels,
+    error: visualNovelsError,
+  } = useRelatedVisualNovelsQuery(Number(id), visualNovelIds, {
+    enabled: visualNovelIds.length > 0,
+  });
+
+  if (traitsError) {
+    return <Error error={traitsError} />;
+  }
+
+  if (visualNovelsError) {
+    return <Error error={visualNovelsError} />;
+  }
+
+  const characterInstances = data && data.instances.length > 0 && (
+    <TagBlock
+      title="Character instances"
+      tags={data.instances.map(instance => ({
+        name: instance.name,
+        note: instance.originalName,
+        path: `/character/${instance.id}`,
+      }))}
+    />
+  );
+
+  const relatedVisualNovels = data && data.visualNovels.length > 0 && visualNovels && visualNovels.length > 0 && (
+    <TagBlock
+      title="Visual novels"
+      tags={visualNovels.map(vn => ({
+        name: vn.title,
+        note: CharacterRole.toReadable(
+          data.visualNovels.find(characterVn => characterVn.visualNovelId === vn.id)?.role ?? CharacterRole.Appears,
+        ),
+        path: `/vn/${vn.id}`,
+      }))}
+    />
+  );
+
+  return (
+    <ContentWrapper isLoading={isLoading || isTraitsLoading || isVisualNovelsLoading} error={error}>
+      {data && (
+        <Container maxW="1440px">
+          <Grid gridTemplateColumns="var(--chakra-sizes-48) 1fr" gap="8" pt="8" mx="auto">
```

Comment:
Please move padding to parent.
All space related rules should be on farthest parent, so when someone would want to change it, they wouldn't need to dig deep into element tree.

Also why you need `mx="auto"`?